caldav/server.go

```diff
@@ -499,8 +500,9 @@ func (b *backend) Put(r *http.Request) (*internal.Href, error) {
 	}
 
 	opts := PutCalendarObjectOptions{
+		// TODO according to spec, If-None-Match can also be an ETag
 		IfNoneMatch: r.Header.Get("If-None-Match") == "*",
-		IfMatch:     r.Header.Get("If-Match"),
+		IfMatch:     strings.Trim(r.Header.Get("If-Match"), "\""),
```

Comment:
We already have `internal.ETag` to parse these.

Reply:
Indeed. Added a helper function for conditional headers that uses it.